packages/components/src/tab-panel/index.tsx

```diff
@@ -136,6 +136,7 @@ export function TabPanel( {
 					role="tabpanel"
 					id={ `${ selectedId }-view` }
 					className="components-tab-panel__tab-content"
+					tabIndex={ -1 }
```

Comment:
> Include [tabindex="-1"](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/tabindex) on enable a tabpanel to receive focus without putting the tabpanel in the page Tab focus sequence for keyboard users.

According to [tabpanel role docs](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tabpanel_role#:~:text=The%20tabpanel%20role%20is%20the,the%20unselected%20tabs%20are%20hidden.), this seems to be okay I think. -cc @ciampo

Reply:
I must admit that I'm not super expert about `TabPanel` (the component pre-dates the start of my work on components, and I haven't really used it often so far), but the MDN docs seem quite clear.

What's the reason why you're adding that attribute specifically for this PR ? Do we want all `TabPanel` instances to have `tabIndex = -1` ?

Best thing to do here would be to split changes to `TabPanel` to a separate PR, so that it can be discussed more in details

Reply:
The reason would be:
>I think the issue is the roving tabindex is applied to the tab content, not the sidebar itself.

I'm not really sure if that's what we want either here or for the component and right now I'm reading and exploring. If we need a change like that eventually, I'll open a separate PR.

In general is really hard because different browsers have different behaviors for me right now and is baffling 😄

Reply:
Opened an issue to track this separately https://github.com/WordPress/gutenberg/issues/45390

Reply:
@ntsekouras That is not the reason. The reason is, your PR changes the DOM order. In trunk, I can press tab once focus is placed in the search input and move in to the `TabPanel`. Because the focus is placed in the search input and the search input is wrapped with the constrained tabbing as part of the inserter, there is no way to get back to the tabs as those seem to be outside of the constrained tabbing area.

This is one of those things that needs to be fixed before your PR gets merged or we will end up making the inserter inaccessible. :(

If you do a fresh build of trunk, try the following.
1. Open the inserter.
2. Press tab.
3. Focus is placed on the Blocks tab item as normal.

In your PR.
1. Open the inserter.
2. Notice how the search input receives focus but it is below the tabs in the DOM.
3. Press shift+tab to move to the Blocks tab.
4. You land at the bottom category of the inserter.

Reply:
Ah, never mind. I think it will not matter for this PR. You can eventually get to the tabs, but it is a little hard.

Can someone open a follow-up issue to apply the constrained tabbing to the inserter sidebars and not the specific components themselves? This should help out a lot with this. That way the sidebar has the same roving tabindex no matter what tab is selected.

Aside from that, need to wait until the above issue for manual tab activation is implemented. Once that is done, I give A11Y sign off on this.

Reply:
@alexstine I've opened [manual tab activation PR](https://github.com/WordPress/gutenberg/pull/45411). It's still not 100% clear to me though what else I should do regarding the constraint tabbing issue you mention. Inserter sidebars use the `useConstrainedTabbing` hook and for me it seems we have an existing issue in trunk with composite navigation in some browsers like firefox.

Example:
1. Open inserter and leave it to blocks tab
2. Focus goes to search input
3. Try to enter the block types list with `tab`
4. Observe that you can't.

Do you have some specific solution in mind code wise?

Reply:
@ntsekouras Not at this point. I just do not understand how it works right now. I am busy working on other issues.